src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -23,6 +23,8 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
         {
             IEnvironment environment = builder.ApplicationServices.GetService<IEnvironment>() ?? SystemEnvironment.Instance;
 
+            builder.UseMiddleware<EnvironmentReadyCheckMiddleware>();
+
```

Comment:
Had to move this check up. In the assignment phase, we call DelayRequests(). But if a call comes in while delayed, it will skip our PlaceholderSpecializationMiddleware and end up returning a 404. By pausing here, it will correctly specialize and find the function as expected.